erpnext/hr/doctype/employee_transfer/employee_transfer.py

```diff
@@ -29,14 +47,12 @@ def on_submit(self):
 				new_employee.internal_work_history = []
 				new_employee.date_of_joining = self.transfer_date
 				new_employee.company = self.new_company
-			#move user_id to new employee before insert
 			if employee.user_id and not self.validate_user_in_details():
 				new_employee.user_id = employee.user_id
 				employee.db_set("user_id", "")
 			new_employee.insert()
 			self.db_set("new_employee_id", new_employee.name)
-			#relieve the old employee
-			employee.db_set("relieving_date", self.transfer_date)
+			employee.db_set("relieving_date", self.to_date if self.to_date else self.transfer_date)
```

Comment:
revert: do not allow `create_new_employee_id` for temporary transfers (validate it as well)